arch/sim/src/sim/up_allocateheap.c

```diff
@@ -71,6 +75,15 @@ static uint8_t sim_heap[SIM_HEAP_SIZE];
 
 void up_allocate_heap(void **heap_start, size_t *heap_size)
 {
+  /* Note: Some subsystems like modlib and binfmt need to allocate
+   * executable memory. We make the entire heap executable here to keep
+   * the sim simpler. If it turns out to be a problem, the
+   * ARCH_HAVE_MODULE_TEXT mechanism can be an alternative.
+   */
+
+#if defined(CONFIG_LIBC_MODLIB) || defined(CONFIG_BINFMT_LOADABLE)
```

Comment:
How about we change to:
```
#if defined(CONFIG_LIBC_MODLIB) || defined(CONFIG_BINFMT_LOADABLE)
  *heap_start = host_alloc_heap(SIM_HEAP_SIZE);
#else
  static uint8_t sim_heap[SIM_HEAP_SIZE];
  *heap_start = sim_heap;
#endif
```
And remove the line 55-59.

Reply:
will do

Reply:
done